KAFKA-4406: add ssl.provider.classes config option

reftel opened the pull request.

reftel committed:
KAFKA-4406: add ssl.provider.classes config option

harshach commented on clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java:
```diff
@@ -123,7 +133,12 @@ else if (clientAuthConfig.equals("requested"))
+        throws GeneralSecurityException, IOException, ClassNotFoundException, IllegalAccessException,
+        InstantiationException {
+        for (String providerClassName : providerClasses) {
+            Security.addProvider((Provider) Class.forName(providerClassName).newInstance());
```
can you use the Utils.newInstance here . https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L312

harshach reviewed (commented):
overall looks good to me.

reftel commented:
Thanks for your comments! I've updated as you suggested, and also fixed some issues found by checkstyle etc.
Please note, though, that this change is being discussed at https://issues.apache.org/jira/browse/KAFKA-4406 , and should not be merged until the discussion has concluded.

omkreddy commented:
This was implemented in KAFKA-8669
